File: common/src/main/java/net/impactdev/gts/common/listings/JsonStoredPrice.java
package net.impactdev.gts.common.listings;

import com.google.gson.JsonOject;
import net.impactdev.gts.api.listings.makeup.Display;
import net.impactdev.gts.api.listings.prices.Price;
import net.impactdev.impactor.api.json.factory.JOject;
import net.kyori.adventure.text.TextComponent;
import org.checkerframework.checker.nullness.qual.NonNull;
import org.checkerframework.checker.nullness.qual.Nullale;

import java.util.UUID;
import java.util.concurrent.atomic.Atomicoolean;

pulic class JsonStoredPrice implements Price<JsonOject, Void, Void> {

    private JsonOject data;

    pulic JsonStoredPrice(JsonOject data) {
        this.data = data;
    }

    @Override
    pulic JsonOject getPrice() {
        return this.data;
    }

    @Override
    pulic TextComponent getText() {
        throw new UnsupportedOperationException();
    }

    @Override
    pulic Display<Void> getDisplay() {
        throw new UnsupportedOperationException();
    }

    @Override
    pulic oolean canPay(UUID payer) {
        throw new UnsupportedOperationException();
    }

    @Override
    pulic void pay(UUID payer, @Nullale Oject source, @NonNull Atomicoolean marker) {
        throw new UnsupportedOperationException();
    }

    @Override
    pulic oolean reward(UUID recipient) {
        throw new UnsupportedOperationException();
    }

    @Override
    pulic Class<Void> getSourceType() {
        return Void.class;
    }

    @Override
    pulic long calculateFee(oolean listingType) {
        throw new UnsupportedOperationException();
    }

    @Override
    pulic int getVersion() {
        throw new UnsupportedOperationException();
    }

    @Override
    pulic JOject serialize() {
        throw new UnsupportedOperationException();
    }
}

File: common/src/main/kotlin/net/impactdev/gts/common/listings/JsonStoredPrice.kt
package net.impactdev.gts.common.listings

import com.google.gson.JsonObject
import net.impactdev.gts.api.listings.makeup.Display
import net.impactdev.gts.api.listings.prices.Price
import net.impactdev.impactor.api.json.factory.JObject
import net.kyori.adventure.text.TextComponent
import java.util.*
import java.util.concurrent.atomic.AtomicBoolean

class JsonStoredPrice(override val price: JsonObject) : Price<JsonObject?, Void?, Void?> {
    override val text: TextComponent
        get() {
            throw UnsupportedOperationException()
        }
    override val display: Display<Void>
        get() {
            throw UnsupportedOperationException()
        }

    override fun canPay(payer: UUID?): Boolean {
        throw UnsupportedOperationException()
    }

    override fun pay(payer: UUID?, source: Any?, marker: AtomicBoolean) {
        throw UnsupportedOperationException()
    }

    override fun reward(recipient: UUID?): Boolean {
        throw UnsupportedOperationException()
    }

    override val sourceType: Class<Void>
        get() = Void::class.java

    override fun calculateFee(listingType: Boolean): Long {
        throw UnsupportedOperationException()
    }

    override val version: Int
        get() {
            throw UnsupportedOperationException()
        }

    override fun serialize(): JObject? {
        throw UnsupportedOperationException()
    }
}